policy-controller/src/lib.rs

```diff
@@ -136,14 +147,35 @@ impl DiscoverOutboundPolicy<OutboundDiscoverTarget> for OutboundDiscover {
         port: NonZeroU16,
         source_namespace: String,
     ) -> Option<OutboundDiscoverTarget> {
+        // first try to lookup a service that we have indexed
+        if let Some(outbound::ResourceRef { name, namespace }) = self.0.read().lookup_service(addr)
+        {
+            return Some(OutboundDiscoverTarget {
+                kind: TargetKind::Service,
+                name,
+                namespace,
+                port,
+                source_namespace: source_namespace.clone(),
+            });
+        }
+
+        // next, check if a pod with this IP exists. If it does, return a None
+        // so we can let the destinations controller serve the discovery request
+        // as usual
+        if self.0.read().pod_exists(addr) {
+            return None;
+        }
```

Comment:
I'm pretty sure that this isn't actually required -- the current contract is that the proxy will always prefer an endpoint response from the profile API before falling back to a policy, right?

Reply:
I will add some more comments explaining why it is required.

Reply:
I have added some comments explaining why we do that and why it is needed: https://github.com/linkerd/linkerd2/pull/13150/commits/5e6db43096119e32c2cbe7649a83aa57f71c0742